django_irods/storage.py

```diff
@@ -372,7 +372,27 @@ def size(self, name):
         if "CAT_NO_ROWS_FOUND" in stdout:
             raise ValidationError("{} cannot be found in iRODS to retrieve "
                                   "file size".format(name))
-        return int(stdout)
+        return int(float(stdout))
+
+    def checksum(self, name):
+        # first force checksum (re)computation
+        self.session.run("ichksum", None, "-f", name)
+        # retrieve checksum using iquest
+        if '/' in name:
+            file_info = name.rsplit('/', 1)
+            obj_name = file_info[1]
+        else:
+            obj_name = name
+        qrystr = "SELECT DATA_CHECKSUM WHERE DATA_NAME = '{}'".format(obj_name)
+        stdout = self.session.run("iquest", None, "%s", qrystr)[0]
+        if "CAT_NO_ROWS_FOUND" in stdout:
+            raise ValidationError("{} cannot be found in iRODS to retrieve "
+                                  "checksum".format(obj_name))
+        # remove potential '\n' from stdout
+        if stdout.endswith('\n'):
```

Comment:
why not use rstrip instead of if/else?  Why is stdout cast to a str below?  It looks like a string already from the code on this line.

Reply:
Good idea to use rstrip. Agreed str cast is not necessary here.